app/delivery/send_to_providers.py

```diff
@@ -146,12 +177,16 @@ def send_email_to_provider(notification: Notification):
         for key in file_keys:
             check_file_url(personalisation_data[key]["document"], notification.id)
             sending_method = personalisation_data[key]["document"].get("sending_method")
-
+            direct_file_url = personalisation_data[key]["document"]["direct_file_url"]
+            document_id = personalisation_data[key]["document"]["id"]
+            scan_verdict_response = document_download_client.check_scan_verdict(service.id, document_id, sending_method)
+            check_for_malware_errors(scan_verdict_response.status_code, notification)
+            current_app.logger.info(f"scan_verdict for document_id {document_id} is {scan_verdict_response.json()}")
             if sending_method == "attach":
                 try:
-
-                    req = urllib.request.Request(personalisation_data[key]["document"]["direct_file_url"])
+                    req = urllib.request.Request(direct_file_url)
                     with urllib.request.urlopen(req) as response:
+                        check_for_malware_errors(response.getcode(), notification)
```

Comment:
I am confused why we call `check_for_malware_errors` again and with the status code of the file download link. Maybe  talking offline would make this faster with this review.

Reply:
I had intended this as an extra check for safety, since requesting to download a file tagged malware will also return an error code once that is implemented over on the dd-api.

But I agree it is confusing to have the check twice and I can't think of a case where the first check would pass and the 2nd check would fail. So I'll just remove this.